File: Back/src/main/java/com/mutpio/pfe/Entities/Prospect.java
package com.mutpio.pfe.Entities;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.antlr.v4.runtime.misc.NotNull;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Prospect implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long idProspect;

    String nom, prenom;

    LocalDate dateNaissance;

    @Enumerated(EnumType.STRING)
    Regime regime = Regime.GENERALE;

    boolean PPE;

    @OneToMany(mappedBy = "prospect")
    @JsonIgnore
    Set<Devis> devis;

    @OneToOne(mappedBy = "prospect" )
    @JsonIgnore
    Compte compte;

    @OneToOne(mappedBy = "prospect")
    Contact contact;

    @OneToOne(mappedBy = "prospect")
    AdressePostale adressePostale ;

}

File: Front/mutpio/src/app/Models/Prospect.ts
import {Devis} from "./Devis";
import {Compte} from "./Compte";
import {Contact} from "./Contact";
import {AdressePostale} from "./AdressePostale";
import {Regime} from "./Regime";

export class Prospect {
  idProspect!: number;
  nom!: string;
  prenom!: string;
  dateNaissance!: Date;
  regime!: Regime;
  PPE!: boolean;
  devis!: Devis[];
  compte!: Compte;
  contact!: Contact;
  adressePostale!: AdressePostale;
}
